source/include/FreeRTOS_IP_Common.h

```diff
@@ -47,6 +47,14 @@ typedef union IP_Address
     IPv6_Address_t xIP_IPv6; /**< IPv6 address */
 } IP_Address_t;
 
+/** @brief A struct that can hold either an IPv4 or an IPv6 address. */
+typedef struct xxIPv46_Address
+{
+    /* A struct that can hold either an IPv4 or an IPv6 address. */
+    IP_Address_t xIPAddress; /**< IP address contains either IPv4 or IPv6. */
+    BaseType_t xIs_IPv6;     /**< pdTRUE if IPv6 address. */
+} IPv46_Address_t;
```

Comment:
Hein's structure is used in DNS files extensively and we are making it a union now by using IP_Address_t in the same strcture, Please check it is not breaking anything in DNS files.

Reply:
Already updated in this PR. Please check the change in DNS_Cache.c & DNS_Parser.c.
And I can pass unit test after changing.